### PR TITLE
Support both ghc-8.10 and ghc-9.2

### PR DESCRIPTION
**Pull request by erikd**

### CODE REVIEW

**Comment by erikd** [comment]
This should not be merged until the temp fixes in commit "cabal.project : Temp fixes" are sorted out.

**Comment by erikd** [comment]
@tdammers I would like to particularly point you to changes I had to make to the crypto code.

File: cardano-crypto-class/src/Cardano/Crypto/Util.hs [hunk 1]
```diff
@@ -132,12 +141,22 @@ bytesToInteger (BS.PS fp (GHC.I# off#) (GHC.I# len#)) =
...
         -- endian encoding.
-        in GMP.importIntegerFromAddr addrOff# (GHC.int2Word# len#) 1#
+        in importIntegerFromAddr addrOff# (GHC.int2Word# len#) 1#
+  where
+    importIntegerFromAddr :: Addr# -> Word# -> Int# -> IO Integer
+#if __GLASGOW_HASKELL__ >= 900
+-- Use the GHC version here because this is compiler dependent, and only indirectly lib dependent.
+    importIntegerFromAddr addr sz endian = IO $ \s ->
+       case integerFromAddr# sz addr endian s of
+          (# s', i #) -> (# s', i #)
```
Comment (lehins) [inline cardano-crypto-class/src/Cardano/Crypto/Util.hs:156]: Why not?
```suggestion
    importIntegerFromAddr = integerFromAddr
```

File: cardano-crypto-class/src/Cardano/Crypto/Util.hs [hunk 2]
```diff
@@ -43,14 +45,21 @@ import           Data.ByteString (ByteString)
...
-import qualified GHC.IO      as GHC (unsafeDupablePerformIO)
 import qualified GHC.Natural as GHC
-import qualified GHC.Integer.GMP.Internals as GMP
 import           Foreign.ForeignPtr (withForeignPtr)
 
 import           Crypto.Random (MonadRandom (..))
 
+#if __GLASGOW_HASKELL__ >= 900
+-- Use the GHC version here because this is compiler dependent, and only indirectly lib dependent.
+import           GHC.Num.Integer (integerFromAddr#)
```
Comment (lehins) [inline cardano-crypto-class/src/Cardano/Crypto/Util.hs:57]: There is an already matching IO action
```suggestion
import           GHC.Num.Integer (integerFromAddr)
```

File: heapwords/src/Cardano/HeapWords.hs [hunk 3]
```diff
@@ -399,7 +407,23 @@ instance HeapWords Bool where
...
+    -- > IS !Int#
+    --
+    -- so @(IS !Int)@ requires:
+    --
+    -- - 1 word for the 'IS' object header
+    -- - 1 word for the single 'IS' unboxed field, of type 'Int#'
+    --
+    -- ┌──┬──────┐
+    -- │S#│ Int# │
+    -- └──┴──────┘
```
Comment (lehins) [inline heapwords/src/Cardano/HeapWords.hs:423]:
```suggestion
    -- ┌──┬──────┐
    -- │IS│ Int# │
    -- └──┴──────┘
```

**Review by lehins — APPROVED** [review]
Awesome stuff. Thank you!

**Pull request merged**